lib/internal/repl.js

```diff
@@ -119,7 +119,7 @@ function setupHistory(repl, historyPath, oldHistoryPath, ready) {
     }
 
     if (data) {
-      repl.history = data.split(/[\n\r]+/, repl.historySize);
+      repl.history = data.split(/[\n\r]+/).slice(repl.historySize * -1);
```

Comment:
@Fishrock123 Since the history file is going to be arranged from least recent to most (most being at the bottom), doing `data.split(/[\n\r]+/, repl.historySize)` will end up losing the latest history elements if `data.length > repl.historySize`. The latest history elements will end up getting dropped/truncated as they are at a higher index in `data` array than the less recent ones.